### ISSUE TITLE
CustomAttributes not returning all attributes??

### ISSUE THREAD

**Issue by i00** [jbevain/cecil#826]
I have a class with 2 custom attributes:
![image](https://user-images.githubusercontent.com/1435262/151760185-0061938e-3c17-4041-8838-0aba246940b2.png)
But Mono.Cecil is only picking up one:
![image](https://user-images.githubusercontent.com/1435262/151760437-0ca4c132-9b36-4ee2-83ce-6d1e8f10a9bb.png)

**Comment by SimonCropp** [comment]
perhaps share a failing unit test as a Pull Request

**Comment by jbevain** [comment]
I think [Serializable] is compiled as a flag on the TypeDefinition, not as a Custom Attribute.

**Comment by Zastai** [comment]
Exactly. Some attributes are intended for and consumed by the compiler without also being retained as an attribute.

Another example would be the `[MethodImpl]` attribute, which is used to mode some flags that can be specified on a method in IL.
I added 2 attributes on a method in Cecil; `[Obsolete]` shows up as an attribute in a `.custom` declaration; the latter merely becomes the `noinlining` keyword on the method signature. I suppose it's the difference with a CIL attribute and a _custom_ attribute, both of which can be managed via a C# `Attribute`.
![image](https://user-images.githubusercontent.com/13708975/173188863-0aa8c35f-3f44-4850-861b-c717ce0760f0.png)
